### PR TITLE
Fixed long start up time and improved build process (fixes #52)

### LINKED ISSUE #52: VNC startup time long for ROS 2
The `/startup-vnc.sh` for the ROS 2 container takes a long time to start. This is because it has to do `chown -R root:root /root` as the first command.

With DDS installed, a fair number of files (~3500) are in the `/root/rti_workspace` directory now. This causes the chown command to take about 30 seconds to complete.

Two options: 

1. Move the chown command to the Dockerfile for ROS2 (although all files in` /root` seem to be owned by `root` already)
2. Find a way to move the location of the `rti_workspace`

### PR DESCRIPTION
**Pull request by schmerl**
This pull request fixes #52 in a couple of ways:

1. Explicitly install rti-connext and copy a configuration file that changes the workspace location and prevents copying of the examples
2. Moves `chown -R root:root /root` to Dockerfile
3. Separates out the large build step that installs apps and runs the prebuild script into two distinct steps (which saves time when building and only changes the prebuild step)
4. Reorganizes the ros2 docker files so that scripts required by the prebuild script are in `pbfiles/`

These last two changes ease some  building problems (i.e., if only changes the build script or the files associated with it, you had to, in the previous version, build everything in pkgs.rosinstall. This would take about 1 hour to build).

### CODE REVIEW

File: ros2/Dockerfile [hunk 1]
```diff
@@ -85,21 +86,19 @@ ENV TINI_VERSION v0.9.0
...
-
 RUN apt-get update \
   && apt install -y \
     libgazebo9-dev \
     ros-${ROS_DISTRO}-gazebo-ros-pkgs \
+    rti-connext-dds-${RTI_VERSION} \
   && apt-get clean \
   && rm -rf /var/lib/apt/lists/*
+# Make the default workspace somewhere other than /root, and do not copy the examples
+RUN mv /rticommon_config.sh /opt/rti.com/rti_connext_dds-${RTI_VERSION}/resource/scripts/rticommon_config.sh
```
Comment (ChrisTimperley) [inline ros2/Dockerfile:97]: We can just fold this into the `RUN` command above.

File: ros2/Dockerfile [hunk 2]
```diff
@@ -85,21 +86,19 @@ ENV TINI_VERSION v0.9.0
...
 
 # build package
 ARG DIRECTORY
-COPY "${DIRECTORY}" /.dockerinstall
+COPY "${DIRECTORY}"/pkgs.rosinstall "${DIRECTORY}"/apt.list* /.dockerinstall/
```
Comment (ChrisTimperley) [inline ros2/Dockerfile:101]: "${DIRECTORY}/pkgs.rosinstall"
"${DIRECTORY}/apt.list"

Comment (ChrisTimperley) [reply ros2/Dockerfile:101]: We probably want to fix these quotes and add a comment above this line that explains the clever hack that's going on.

File: ros2/Dockerfile [hunk 3]
```diff
@@ -112,10 +111,18 @@ RUN . /opt/ros/${ROS_DISTRO}/setup.sh \
       --ignore-src \
       --skip-keys="python-rosdep python-catkin-pkg python-rospkg" \
       --rosdistro="${ROS_DISTRO}" \
- && (test -f /.dockerinstall/prebuild.sh \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+# We copy pkgs.rosinstall again because copy needs at least one file that exists
+# But we explicitly copy prebuild.sh and files that are used by it in pbfiles
+COPY "${DIRECTORY}"/pkgs.rosinstall  "${DIRECTORY}"/pbfiles* "${DIRECTORY}"/prebuild.sh* /.dockerinstall/
```
Comment (ChrisTimperley) [inline ros2/Dockerfile:119]:
```
COPY "${DIRECTORY}" /.dockerinstall
```

File: ros2/Dockerfile [hunk 4]
```diff
@@ -112,10 +111,18 @@ RUN . /opt/ros/${ROS_DISTRO}/setup.sh \
...
+RUN apt-get update\
+     && (test -f /.dockerinstall/prebuild.sh \
      && (echo "running prebuild step..." && /bin/bash /.dockerinstall/prebuild.sh || exit 1) \
```
Comment (ChrisTimperley) [inline ros2/Dockerfile:122]: Indent 122 and 123 to indicate nested conditionals.

Comment (schmerl) [reply ros2/Dockerfile:125]: Done

File: ros2/Dockerfile (hunk 3 above)
Comment (schmerl) [reply ros2/Dockerfile]: Done

File: ros2/Dockerfile (hunk 2 above)
Comment (schmerl) [reply ros2/Dockerfile:106]: Done

File: ros2/Dockerfile (hunk 1 above)
Comment (schmerl) [reply ros2/Dockerfile]: Done

**Pull request merged**